specification/web/Web.Management/AppServiceEnvironmentResource.tsp

```diff
@@ -0,0 +1,503 @@
+   */
+  @summary("Send a test notification that an upgrade is available for this App Service Environment.")
+  testUpgradeAvailableNotification is ArmResourceActionSync<
+    AppServiceEnvironmentResource,
+    void,
+    {
+      @body body: void;
+    },
+    Error = DefaultErrorResponse
+  >;
+
+  /**
+   * Description for Initiate an upgrade of an App Service Environment if one is available.
+   */
+  @summary("Initiate an upgrade of an App Service Environment if one is available.")
+  upgrade is ArmResourceActionAsyncBase<
+    AppServiceEnvironmentResource,
+    void,
+    ArmAcceptedLroResponse,
+    BaseParameters = Azure.ResourceManager.Foundations.DefaultBaseParameters<AppServiceEnvironmentResource>,
+    Error = DefaultErrorResponse
+  >;
+
+  /**
+   * Description for List all currently running operations on the App Service Environment.
+   */
+  @summary("List all currently running operations on the App Service Environment.")
+  @get
+  @action("operations")
+  listOperations is ArmResourceActionSync<
+    AppServiceEnvironmentResource,
+    void,
+    ArmResponse<Operation[]>,
+    Error = DefaultErrorResponse
+  >;
+
+  /**
+   * Description for Get the network endpoints of all outbound dependencies of an App Service Environment.
+   */
+  @summary("Get the network endpoints of all outbound dependencies of an App Service Environment.")
+  @get
+  @action("outboundNetworkDependenciesEndpoints")
+  @list
+  getOutboundNetworkDependenciesEndpoints is ArmResourceActionSync<
+    AppServiceEnvironmentResource,
+    void,
+    ArmResponse<OutboundEnvironmentEndpointCollection>,
+    Error = DefaultErrorResponse
+  >;
+
+  /**
+   * Description for Gets the private link resources
+   */
+  @summary("Gets the private link resources")
+  @get
+  @action("privateLinkResources")
+  getPrivateLinkResources is ArmResourceActionSync<
+    AppServiceEnvironmentResource,
+    void,
+    ArmResponse<PrivateLinkResourcesWrapper>,
+    Error = DefaultErrorResponse
+  >;
+
+  /**
+   * Description for Reboot all machines in an App Service Environment.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/lro-location-header" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-post-operation-response-codes" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  @summary("Reboot all machines in an App Service Environment.")
+  reboot is ArmResourceActionSync<
+    AppServiceEnvironmentResource,
+    void,
+    AcceptedResponse,
+    Error = DefaultErrorResponse
+  >;
+
+  /**
+   * Description for Resume an App Service Environment.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/no-response-body" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  @summary("Resume an App Service Environment.")
+  @list
+  resume is ArmResourceActionAsyncBase<
```

Comment:
This operation is lro+paging,  Js currently does not support.  The same operation also applies to `suspend` and  `changeVnet`. issue track: https://github.com/Azure/autorest.typescript/issues/2313